src/main/java/de/internetsicherheit/brl/bloxberg/cache/Launcher.java

```diff
@@ -2,10 +2,57 @@
 
 import de.internetsicherheit.brl.bloxberg.cache.gui.MainGUI;
 
+import java.io.IOException;
+
 public class Launcher {
 
-    public static void main(String[] args) {
-        MainGUI.main(args);
+    public static void main(String[] args) throws IOException {
+        String[] parsedArgs = parseArgs(args);
+        if (parsedArgs[4].equals("gui")) {
+            MainGUI.main(args);
+        } else {
+            LogicController logicController = new LogicController();
+            logicController.createJson(parsedArgs);
+        }
+
     }
 
+
+    private static String[] parseArgs(String[] args) {
```

Comment:
While it's function-wise okay to have this as a static function in the launcher, the much cleaner approach would be having this in a separate class, that also returns a structured result object. This would allow for it to be easily unit testable.